src/core/include/openvino/core/layout.hpp

```diff
@@ -129,6 +130,21 @@ OPENVINO_API bool has_width(const Layout& layout);
 ///
 OPENVINO_API std::int64_t width_idx(const Layout& layout);
 
+/// \brief Sets Layout of port
+///
+/// \throws ov::Exception if port is not connected with Result or Parameter
+OPENVINO_API void set_layout(ov::Output<ov::Node> output, const ov::Layout& layout);
```

Comment:
what if I `set_layout` to the node before the `Result`? I suppose it should work